pkg/notifications/stream.go

```diff
@@ -12,7 +12,8 @@ type Stream interface {
 	Produce(event *storage.Notification) error
 }
 
-func newStream() Stream {
+// NewStream creates a new notification stream.
+func NewStream() Stream {
```

Comment:
I guess I'll the reason in the following PRs but here it's unclear why `newStream()` should become exported. Please strive to colocate relevant changes in one PR.

Reply:
Reverted the change, it only was required because it was used in a test in a later PR.

Refactored it to use a separate exported function like `GetTestStream(t *testing.T)` to only allow public usage in tests.